src/com/minseungkwon/datastructure/Graph/Graph.java

```diff
@@ -44,6 +44,7 @@ public String dfs() {
     }
 
     public String dfs(int index) {
+        initMarks();
```

Comment:
I think this is still side-effect method. It would be better to pull out the marked variable from the node, and manage the marked status in each methods.